### PR TITLE
:ticket: ai/server communication start

### PR DESCRIPTION
**Pull request by landryarki**
see  #22.

the server and the client will start their comms as so:
<-- WELCOME \ n
--> TEAM - NAME \ n
<-- CLIENT - NUM \ n
<-- X Y \ n

atm this whole process is blocking...

### CODE REVIEW

File: Network/Makefile_Network.mk [hunk 1]
```diff
@@ -68,7 +68,7 @@ TESTS_FLAGS		= $(TESTS_INCLUDE) $(WARNINGS) $(TESTS_LIBS)
 all:	$(NAME)
 
 $(NAME):	$(OBJ)
-	@gcc -o $(NAME) $(OBJ) $(CFLAGS)
+	@gcc -o $(NAME) $(OBJ) $(CFLAGS) $(VALGRIND)
```
Comment (Antoinegtir) [inline Network/Makefile_Network.mk:71]: remove Valgrind flag

File: Network/include/Server/server.h [hunk 2]
```diff
@@ -42,22 +42,12 @@ typedef enum {
...
-    struct sockaddr_in *clientAddress;
-} client_server_t;
-
 /**
  * @struct client_s
  * @brief Structure representing a client.
  */
 typedef struct client_s {
-    client_server_t *clientServer;
+    int socket;
```
Comment (Antoinegtir) [inline Network/include/Server/server.h:50]: no needed to create a struct to store only one data type?

**Review by Antoinegtir — CHANGES_REQUESTED** [review]

File: Network/include/Server/server.h (hunk 2 above)
Comment (landryarki) [reply Network/include/Server/server.h:50]: things will be added later mb

File: Network/Makefile_Network.mk (hunk 1 above)
Comment (landryarki) [reply Network/Makefile_Network.mk:71]: why should i ?

Comment (Peralban) [reply Network/Makefile_Network.mk:71]: comment the valgrind flag

Comment (landryarki) [reply Network/Makefile_Network.mk:71]: again why should i ?

File: Network/src/Parsing/move_args_to_struct.c [hunk 3]
```diff
@@ -34,26 +33,33 @@ bool complete_integer_data(server_t *server, char **args)
...
-    init_game(server->info_game) == NULL) {
-        if (server->info_game.team_names != NULL)
-            my_free_array(server->info_game.team_names);
-        free(server);
-        free(args);
-        return NULL;
-    }
-    free(args);
-    return server;
+    return get_s(args, server, return_null);
```
Comment (Peralban) [inline Network/src/Parsing/move_args_to_struct.c:64]: This name isn't very precise.

**Review by Peralban — CHANGES_REQUESTED** [review]

**Review by Peralban — APPROVED** [review]

File: Network/Makefile_Network.mk (hunk 1 above)
Comment (Antoinegtir) [reply Network/Makefile_Network.mk:71]: because the flag compiles the program in a less optimized way than if it were removed

**Pull request merged**